sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -159,37 +160,17 @@ public HttpTransport(
     int responseCode = -1;
 
     try (final OutputStream outputStream = connection.getOutputStream();
-        final Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream, UTF_8))) {
+        final GZIPOutputStream gzip = new GZIPOutputStream(outputStream);
+        final Writer writer = new BufferedWriter(new OutputStreamWriter(gzip, UTF_8))) {
```

Comment:
TIL, I didn't know this "try-with-resources" https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html 👍 

```java
try (...) { ... }
```